hub/util/path.py

```diff
@@ -12,3 +14,40 @@ def get_path_from_storage(storage):
             return storage.hub_path
         return storage.root
     return None
+
+
+def find_root(path: str) -> str:
+    # TODO: update docstring
+    # TODO: tests
+
+    """
+    Find the root of the dataset within the given path.
+    the "root" is defined as being the path to a subdirectory within path that has > 1 folder/file (if applicable).
+    in other words, if there is a directory structure like:
+    dataset -
+        Images -
+            class1 -
+                img.jpg
+                ...
+            class2 -
+                img.jpg
+                ...
+            ...
+    the output of this function should be "dataset/Images/" as that is the root.
+    """
+
+    subs = glob.glob(os.path.join(path, "*"))
+
+    # if ignore_sub:
+    #     ignore_sub = os.path.join(path, ignore_sub)
+    #     if ignore_sub in subs:
+    #         subs.remove(ignore_sub)
+
```

Comment:
could you redo this docstring? it basically just needs to be refactored and updated.

also, can you remove the commented out code? 

you can remove the "test" TODO, it is already covered in your tests :-P